python/lsst/daf/butler/formatters/yamlFormatter.py

```diff
@@ -54,13 +54,48 @@ def _readFile(self, path, pytype=None):
         The `~yaml.UnsafeLoader` is used when parsing the YAML file.
         """
         try:
-            with open(path, "r") as fd:
-                data = yaml.load(fd, Loader=yaml.UnsafeLoader)
+            with open(path, "rb") as fd:
+                data = self._fromBytes(fd.read())
         except FileNotFoundError:
             data = None
 
         return data
 
+    def _fromBytes(self, inMemoryDataset, pytype=None):
+        """Read the bytes object as a python object.
+
+        Parameters
+        ----------
+        inMemoryDataset : `bytes`
+            Bytes object to unserialize.
+        pytype : `class`, optional
+            Not used by this implementation.
+
+        Returns
+        -------
+        data : `object`
+            Either data as Python object read from the pickled string, or None
+            if the string could not be read.
+        """
+        try:
+            data = yaml.load(inMemoryDataset, Loader=yaml.UnsafeLoader)
+        except yaml.YAMLError:
+            data = None
+        try:
+            data = data.exportAsDict()
+        except AttributeError:
+            # its either my mis-use of yaml or intended behaviour, but yaml
```

Comment:
Don't worry about it. Yes, the coerceType later on is trying to fix the fact that generic YAML files don't have type information but the butler really wants some python type other than a dict to be returned in many cases. There is some hackery involved to get things working properly but that's not relevant to S3.